### PR TITLE
Add delete on cascade to fk_svc_inst_op_svc_instance_id

### PR DESCRIPTION
**Pull request by kathap**
We can see double entries for the same service_instance_id
in the service_instance_operations table. This leads to:
PG::ForeignKeyViolation: ERROR:  update or delete on table
"service_instances" violates foreign key constraint
"fk_svc_inst_op_svc_instance_id" on table "service_instance_operations"

For service bindings, service keys and route bindings the foreign
key constraint from operation to resource contains ON DELETE CASCADE;
for service instances this is missing.

Adding ON DELETE CASCADE to FK fk_svc_inst_op_svc_instance_id in
service_instance_operations

* Links to any other associated PRs
The root cause is fixed in this PR: #2916
#2929

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [X] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

File: db/migrations/20220805145100_add_delete_on_cascade_to_fk_svc_inst_op_svc_instance_id.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Sequel.migration do
+  # Add delete cascade to fk_svc_inst_op_svc_instance_id, without there won't be cleaned up all ocurrencies in
```
Comment (philippthun) [inline db/migrations/20220805145100_add_delete_on_cascade_to_fk_svc_inst_op_svc_instance_id.rb:2]: I find this sentence hard to read. What about `Add DELETE CASCADE to foreign key fk_svc_inst_op_svc_instance_id to ensure that 'service instance operations' are deleted together with the referenced 'service instance'.`?

File: spec/unit/models/services/service_instance_spec.rb [hunk 2]
```diff
@@ -183,6 +183,31 @@ module VCAP::CloudController
         expect(ServiceInstanceAnnotationModel.where(id: annotation.id)).to be_empty
       end
 
+      it 'cascade deletes all ServiceInstanceOperations for this instance' do
```
Comment (philippthun) [inline spec/unit/models/services/service_instance_spec.rb:186]: In theory there should only be a single operation - and this test also instantiates a single operation, thus I would change the text to `it 'cascade deletes the related ServiceInstanceOperation for this instance'`.

File: spec/unit/models/services/service_instance_spec.rb [hunk 3]
```diff
@@ -183,6 +183,31 @@ module VCAP::CloudController
...
+        last_operation = ServiceInstanceOperation.make
+        service_instance.service_instance_operation = last_operation
+
+        service_instance.destroy
+
+        expect(ServiceInstance.find(guid: service_instance.guid)).to be_nil
+        expect(ServiceInstanceOperation.find(guid: last_operation.guid)).to be_nil
+      end
+
+      it 'cascades deletion of related dependencies' do
```
Comment (philippthun) [inline spec/unit/models/services/service_instance_spec.rb:196]: I would remove this test as it is unrelated to the changes in this PR.

**Review by philippthun — COMMENTED** [review]
The newly added test passes even without the migration. The reason for this is the following line in `app/models/services/service_instance.rb`:
```
add_association_dependencies service_instance_operation: :destroy
```

I would propose to remove this 'association dependency' (then the test should fail without the new migration). This would mean that service instance operations would be consistent with service binding operations, route binding operations and service key operations - they also do not have such an 'association dependency' and use DELETE CASCADE instead.

File: db/migrations/20220805145100_add_delete_on_cascade_to_fk_svc_inst_op_svc_instance_id.rb (hunk 1 above)
Comment (kathap) [reply db/migrations/20220805145100_add_delete_on_cascade_to_fk_svc_inst_op_svc_instance_id.rb:2]: ok, sounds good.

File: spec/unit/models/services/service_instance_spec.rb (hunk 2 above)
Comment (kathap) [reply spec/unit/models/services/service_instance_spec.rb:186]: ok

**Review by philippthun — APPROVED** [review]

**Pull request merged**